### PR TITLE
Be quieter in expected shutdown cases

### PR DESCRIPTION
**Pull request by mikeshardmind**
### Type

- [ ] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes

- shutdown/restart commands are less noisy.
- quitting with keyboard interrupt is warned against, then handled as best as possible.

### CODE REVIEW

**Review by Drapersniper — DISMISSED** [review]
LGTM, also fixed spam in logs i was seeing during a restart

**Comment by Drapersniper** [comment]
During graceful shudowns shudowns are a lot faster on windows.

Forced shutdowns are significantly faster than before too however some times if python is interacting with a task it will throw a trace after shutting the bot down.

image


This is not the case 100% of the time, however we are warning the user with `Please do not use Ctrl+C to Shutdown Red! (attempting to die gracefully...)`  as its an attempt for a graceful shutdown and they are warned to not use `Ctrl+C` so this PR look fine if we are okay with the traces being raise some times on Windows.

**Review by Drapersniper — DISMISSED** [review]

**Comment by Jackenmen** [comment]
I still don't have a graceful shutdown with this PR, getting this when using `[p]shutdown` with non-0 exit code:
```
ExitCodes.SHUTDOWN
CRITICAL:root:Caught unhandled exception: ExitCodes.SHUTDOWN
```

File: redbot/__main__.py [hunk 1]
```diff
@@ -345,31 +345,56 @@ def handle_early_exit_flags(cli_flags: Namespace):
...
+    """
+    Logs unhandled exceptions in other tasks
+    (to standard error, not logs, this applies to all tasks and futures)
+    """
     msg = context.get("exception", context["message"])
-    if isinstance(msg, KeyboardInterrupt):
-        # Windows support is ugly, I'm sorry
-        logging.error("Received KeyboardInterrupt, treating as interrupt")
-        signal_type = signal.SIGINT
+    if isinstance(msg, KeyboardInterrupt, SystemExit):
```
Comment (Jackenmen) [inline redbot/__main__.py]:
```suggestion
    if isinstance(msg, (KeyboardInterrupt, SystemExit)):
```

Comment (mikeshardmind) [reply redbot/__main__.py]: Thanks for catching that one

**Comment by mikeshardmind** [comment]
Proactor event loop has an issue right now not ignoring already being closed. I'll handle this later tonight, and contribute upstream if it doesnt already have someone assigned.

**Review by Kowlin — APPROVED** [review]
Working as intended.

**Pull request merged**